src/Config.Binder/ConfigurationBinder.cs

```diff
@@ -179,7 +180,21 @@ private static void BindNonScalar(this IConfiguration configuration, object inst
         {
             if (instance != null)
             {
-                foreach (var property in GetAllProperties(instance.GetType().GetTypeInfo()))
+                foreach (var requiredProperty in options.RequiredProperties)
+                {
+                    var propertyName = requiredProperty;
```

Comment:
is this var declaration left over from a previous refactor? Could just use `requiredProperty` below and remove this line

Reply:
Nope because it’s set some line after and we can’t set the foreach variable